### PR TITLE
[#37] [Integrate] As a user, I can see the details of a survey

### LINKED ISSUE #37: [Integrate] As a user, I can see the details of a survey
## Why

- Before taking a survey, users should be able to see the full preview of the details so that they can decide whether or not the survey matching their expectation.
 
## Acceptance Criteria
- Upon tapping the Survey Disclosure button in Home, navigate the user to Survey Details with the selected survey.
`Pass and use the selected survey data fetched in Home.`

- Use the survey title as the Title label
- Use the survey description as the Description label
- Use the survey cover image url as the background image
  - The image is zoomed to 120% anchoring to the top right of the screen
- All unit test passed

### PR DESCRIPTION
**Pull request by nkhanh44**
#37 

## What happened

- Integrate Survey Detail Screen to show data
 
## Insight

- Add unit tests
 
## Proof Of Work

**iPhone 12 promax:**

https://user-images.githubusercontent.com/25881847/185831392-e0800b5a-3c8b-48f8-b86a-75369e3e3604.mp4

**iPhone 8:**

https://user-images.githubusercontent.com/25881847/185831409-4ab6642b-fc7f-4ff2-9bf5-ee7ce4bfa0c5.mp4

### CODE REVIEW

File: SurveyTests/Presentation/Modules/SurveyDetail/SurveyDetailViewSpec.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                surveyDetailView = SurveyDetailView(
+                    viewModel: SurveyDetailViewModel(),
+                    isPresented: .constant(true),
+                    survey: survey
+                )
+            }
+
+            context("has onAppear called") {
+
+                it("return correct response") {
```
Comment (blyscuit) [inline SurveyTests/Presentation/Modules/SurveyDetail/SurveyDetailViewSpec.swift]:
```suggestion
                it("did not call next") {
```

**Review by blyscuit — COMMENTED** [review]
Looks good.

File: Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+struct SurveyDetailViewModel {}
+
+extension SurveyDetailViewModel: ViewModel {
+
+    func transform(_ input: Input) -> Output {
+        let output = Output()
+
+        input.nextToSurveyQuestionTrigger
+            .map { true }
+            .assign(to: \.didNextSuccessfully, on: output)
```
Comment (ducbm051291) [inline Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailViewModel.swift]: How about this? 
```suggestion
            .assign(to: \.willGoToNextSurvey, on: output)
```

File: Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift [hunk 3]
```diff
@@ -53,7 +39,9 @@ struct SurveyDetailView: View {
 
                         SButtonView(
                             isValid: .constant(true),
-                            action: {},
+                            action: {
+                                nextToSurveyQuestionTrigger.send()
```
Comment (ducbm051291) [inline Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift]: The variable is confused here. Should it be startSurveyTrigger? Correct me if I'm wrong, thanks. 
```suggestion
                                startSurveyTrigger.send()
```

File: SurveyTests/Presentation/Modules/SurveyDetail/SurveyDetailViewSpec.swift (hunk 1 above)
Comment (ducbm051291) [reply SurveyTests/Presentation/Modules/SurveyDetail/SurveyDetailViewSpec.swift]: @blyscuit Should we use the present tense here?

File: Survey/Sources/Presentation/Modules/Home/Views/SurveyItemView.swift [hunk 4]
```diff
@@ -14,6 +14,7 @@ struct SurveyItemView: View {
     @State private var fadeInOut = false
 
     let survey: Survey
+    let didNext: () -> Void
```
Comment (ducbm051291) [inline Survey/Sources/Presentation/Modules/Home/Views/SurveyItemView.swift]: I think we should name this `willGoToDetail` or presentSurveyDetail or something else. What  do you think?

**Review by ducbm051291 — COMMENTED** [review]
The rest LGTM. Just need to name variable clearer 😀

**Comment by nkhanh44** [comment]
@ducbm051291 Thank you so much for suggestions, that makes variables more clearer 😄 , I fixed at https://github.com/nkhanh44/ic-survey-ios/pull/43/commits/41d200514af074a2f9211580020f1b4043fe9c52

**Review by blyscuit — APPROVED** [review]

File: SurveyTests/Presentation/Modules/SurveyDetail/SurveyDetailViewSpec.swift (hunk 1 above)
Comment (blyscuit) [reply SurveyTests/Presentation/Modules/SurveyDetail/SurveyDetailViewSpec.swift]: @nkhanh44 You can use `does not call next`

**Review by ducbm051291 — APPROVED** [review]
LGTM

**Pull request merged**